server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/GetFieldMappingsResponse.java

```diff
@@ -93,9 +100,61 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static GetFieldMappingsResponse fromXContent(XContentParser parser) throws IOException {
+        final Map<String, Map<String, Map<String, FieldMappingMetaData>>> mappings = new HashMap<>();
+        final Map<String, Object> map = parser.map();
+
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            final Object value = entry.getValue();
+
+            Map<String, Map<String, FieldMappingMetaData>> typeMappings = new HashMap<>();
+            mappings.put(entry.getKey(), typeMappings);
+
+            if (value instanceof Map) {
+                final Object o = ((Map) value).get(MAPPINGS.getPreferredName());
+                if (!(o instanceof Map)) {
+                    throw new ParsingException(parser.getTokenLocation(), "Nested " + MAPPINGS.getPreferredName() + " is not found");
+                }
```

Comment:
is it necessary to parse everything into a map in the first place? At some point later on we do need a map, but I wonder if we can postpone calling parser.map till then.

Reply:
it's quite good point, will try to do that